### PR TITLE
add release procedures to developer documentation

### PR DESCRIPTION
**Pull request by grlee77**
This adds a document outlining the release procedures for PyWavelets.  This would have been useful to me as a first time release manager, so I made some notes that could hopefully help others in the future.

### CODE REVIEW

File: doc/source/dev/how_to_release.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+number.
+
+
+updating the release notes
+--------------------------
+Prior to the release, make sure the release notes are up to date.
+
+author lists can be generated via:
+
+```python  ./utils/gh_lists.py vP.P.P..```
```
Comment (rgommers) [inline doc/source/dev/how_to_release.rst]: ``util/authors.py``

File: doc/source/dev/how_to_release.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+(where vP.P.P is the previous release number)
+
+and the lists of Issues closed and PRs merged via:
+
+```python ./utils/gh_lists.py vX.X.X```  (script requires Python 2.X to run)
```
Comment (rgommers) [inline doc/source/dev/how_to_release.rst]: ``util`` not ``utils``

File: doc/source/dev/how_to_release.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+Upload the release to pypi
+--------------------------
+
+The source distributions created above can be securely uploaded to
+pypi.python.org using twine:
+
+```twine upload -s dist/*```
+
+Also manually upload the Windows wheels (see above) through the pypi web
+interface.
```
Comment (rgommers) [inline doc/source/dev/how_to_release.rst]: you can also do this with twine. maybe best to first download the Windows wheels into ``dist/`` and then the single run of ``twine upload -s dist/`` will do it all at once.

File: doc/source/dev/how_to_release.rst [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+The documentation on readthedocs will have been automatically generated,
+but the documentation linked from pypi.python.org will need to be manually
```
Comment (rgommers) [inline doc/source/dev/how_to_release.rst]: maybe add "on pythonhosted.org" somewhere here

**Comment by rgommers** [comment]
Very nice. I only had a few minor comments.

**Comment by grlee77** [comment]
okay.
I had also forgotten to mention creating the release on GitHub, so I added that too.

File: doc/source/dev/how_to_release.rst [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+number.
+
+
+updating the release notes
+--------------------------
+Prior to the release, make sure the release notes are up to date.
+
+author lists can be generated via:
+
+```python  ./util/gh_lists.py vP.P.P..```
```
Comment (rgommers) [inline doc/source/dev/how_to_release.rst]: This still says ``gh_lists.py`` instead of ``authors.py``

**Comment by grlee77** [comment]
I think the AppVeyor failure was caused by a temporary glitch

**Comment by rgommers** [comment]
Yep, just a glitch. This looks good now, merging. Thanks @grlee77!

**Comment by aaren** [comment]
This is great. Thanks @grlee77.

**Pull request merged**